### PR TITLE
Authrization  SignIn functionality added

### PR DESCRIPTION
**Pull request by t007rushi**
Check for responsiveness
-> LogIn page with login as guest redirect to notes page

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
+import { HomePage, LoginPage, NotesPage } from "./pages/index";
+import ProtectedRoute from "./Router/ProtectedRoute"
 
 function App() {
   return (
     <Routes>
       <Route path="/" element={<HomePage />}></Route>
       <Route path="/mock" element={<Mockman />}></Route>
+      <Route path="/login" element={<LoginPage />}></Route>
+      {/* <Route path="/signup" element={<SignupPage />}></Route> */}
```
Comment (vijaytembugade) [inline src/App.js:13]: Remove unnecessary code and comments

**Review by vijaytembugade — APPROVED** [review]
Amazing work Rushikesh.
Barrel Exports ,Authentication, and file structure looks really clean.

File: src/App.js (hunk 1 above)
Comment (t007rushi) [reply src/App.js:13]: Yes that route going to implement next👍

File: src/Router/ProtectedRoute.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { useAuth } from "../context/auth-context";
+import { Navigate } from "react-router-dom";
+
+const ProtectedRoute = ({ ProtectedComp }) => {
+  const { user } = useAuth();
+  return (
+    <div>
+      {user.isUserLoggedIn ? ProtectedComp : <Navigate replace to="/login" />}
```
Comment (horsemaker) [inline src/Router/ProtectedRoute.js:9]: Nice implementation of ProtectedRoute! 👌

File: src/context/auth-context.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    <authContext.Provider value={{ user, setUser }}>
+      {children}
+    </authContext.Provider>
+  );
+};
+
+const useAuth = () => {
+  const context = useContext(authContext);
+  if (context === undefined) {
+    throw new Error("useContext must be used inside AuthProvider");
```
Comment (horsemaker) [inline src/context/auth-context.js:37]: Nice safety check! 👌

File: src/pages/Login/LoginPage.jsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+import { Link, useNavigate } from "react-router-dom";
+import "./login.css";
+import { logInHandler } from "../../services";
+import { useAuth } from "../../context/auth-context";
+
+const LoginPage = () => {
+  const { setUser } = useAuth();
+  const [logUser, setLogUser] = useState({
+    email: "",
+    pass: "",
```
Comment (horsemaker) [inline src/pages/Login/LoginPage.jsx]: Maybe mak this `password`. Will be easy to understand.

File: src/pages/index.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import LoginPage from "./Login/LoginPage"
+import HomePage from "./Home/HomePage"
+// import SignupPage from "./Signup/SignupPage"  
+import NotesPage from "./Notes/NotesPage"
+
+export {HomePage,LoginPage,NotesPage}
```
Comment (horsemaker) [inline src/pages/index.js]: You could also do `export { HomePage } from "./Home/HomePage"`

File: src/services/Auth/logInHandler.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import axios from "axios";
+
+//LOGIN
+export const logInHandler = ({ email, pass }, setUser, navigator) => {
```
Comment (horsemaker) [inline src/services/Auth/logInHandler.js]: Since we don't usually use states in services. If you avoid using states in services that would be good.

**Review by horsemaker — APPROVED** [review]

**Pull request merged**